packages/dev/core/src/Misc/snapshotRenderingHelper.ts

```diff
@@ -0,0 +1,228 @@
+     * Default: 20
+     */
+    morphTargetsNumMaxInfluences?: number;
+}
+
+/**
+ * A helper class to simplify work with FAST snapshot mode (WebGPU only - can be used in WebGL too, but won't do anything).
+ */
+export class SnapshotRenderingHelper {
+    private _engine: AbstractEngine;
+    private _scene: Scene;
+    private _options: SnapshotRenderingHelpersOptions;
+    private _onBeforeRenderObserver: Nullable<Observer<Scene>>;
+    private _onBeforeRenderObserverUpdateLayer: Nullable<Observer<Scene>>;
+    private _onResizeObserver: Nullable<Observer<AbstractEngine>>;
+
+    /**
+     * Creates a new snapshot rendering helper
+     * Note that creating an instance of the helper will set the snapshot rendering mode to SNAPSHOTRENDERING_FAST but will not enable snapshot rendering (engine.snapshotRendering is not updated).
+     * Note also that fixMeshes() is called as part of the construction
+     * @param scene The scene to use the helper in
+     * @param options The options for the helper
+     */
+    constructor(scene: Scene, options?: SnapshotRenderingHelpersOptions) {
+        this._scene = scene;
+        this._engine = scene.getEngine();
+
+        if (!this._engine.isWebGPU) {
+            return;
+        }
+
+        this._options = {
+            morphTargetsNumMaxInfluences: 20,
+            ...options,
+        };
+
+        this._engine.snapshotRenderingMode = Constants.SNAPSHOTRENDERING_FAST;
+
+        this.fixMeshes();
+
+        this._onResizeObserver = this._engine.onResizeObservable.add(() => {
+            const save = this._engine.snapshotRendering;
+            this._engine.snapshotRendering = false;
+            if (save) {
+                this.enableSnapshotRendering();
+            }
+        });
+
+        this._scene.onBeforeRenderObservable.add(() => {
+            if (!this._engine.snapshotRendering || this._engine.snapshotRenderingMode !== Constants.SNAPSHOTRENDERING_FAST) {
+                return;
+            }
+
+            // Animate skeletons
+            scene.skeletons.forEach((skeleton) => skeleton.prepare(true));
+
+            for (const mesh of scene.meshes) {
+                if (mesh.infiniteDistance) {
+                    mesh.transferToEffect(mesh.computeWorldMatrix(true));
+                }
+
+                if (mesh.skeleton) {
+                    mesh.transferToEffect(mesh.computeWorldMatrix(true));
+                }
+
+                if (mesh.morphTargetManager && mesh.subMeshes) {
+                    // Make sure morph target animations work
+                    for (const subMesh of mesh.subMeshes) {
+                        const dw = subMesh._drawWrapper;
+                        const effect = dw.effect;
+                        if (effect) {
+                            const dataBuffer = (dw.drawContext as WebGPUDrawContext).buffers[WebGPUShaderProcessor.LeftOvertUBOName];

```

Comment:
With the change earlier where only type info for `WebGPUShaderProcessor` is imported, you can then change this line to:
```suggestion
                            const dataBuffer = drawContext.buffers["LeftOver" satisfies (typeof WebGPUShaderProcessor)["LeftOvertUBOName"]];

```

This makes it so you don't have a runtime/bundler time dependency on `WebGPUShaderProcessor` just for `LeftOverUBOName`, but still does a compile time check (e.g. if you got the `"LeftOver"` string wrong, it would be a compile error).

Alternatively, you could export `LeftOverUBOName` as a module level constant so you don't have to import all of `WebGPUShaderProcessor` (and its depenencies).